crates/loam-cli/src/commands/dev/mod.rs

```diff
@@ -0,0 +1,105 @@
+use clap::Parser;
+use notify::{RecursiveMode, Watcher};
+use std::path::Path;
+use tokio::time;
+
+use crate::commands::build;
+
+use super::build::clients::LoamEnv;
+
+#[derive(Parser, Debug, Clone)]
+#[group(skip)]
+pub struct Cmd {
+    #[command(flatten)]
+    pub build_cmd: build::Cmd,
+}
+
+#[derive(thiserror::Error, Debug)]
+pub enum Error {
+    #[error(transparent)]
+    Watcher(#[from] notify::Error),
+    #[error(transparent)]
+    Build(#[from] build::Error),
+    #[error("IO error: {0}")]
+    Io(#[from] std::io::Error),
+}
+
+impl Cmd {
+    pub async fn run(&self) -> Result<(), Error> {
+        // Set LOAM_ENV to development
+        std::env::set_var("LOAM_ENV", "development");
+        let (tx, mut rx) = tokio::sync::mpsc::channel(100);
+        let mut watcher =
+            notify::recommended_watcher(move |res: Result<notify::Event, notify::Error>| {
+                if let Ok(
+                    event @ notify::Event {
+                        kind: notify::EventKind::Modify(_),
+                        ..
+                    },
+                ) = res
+                {
+                    if let Some(path) = event.paths.first() {
+                        eprintln!("File modified: {path:?}");
+                        // Send a signal through the channel to trigger a rebuild
+                        if let Err(e) = tx.blocking_send(()) {
+                            eprintln!("Error sending through channel: {e}");
+                        }
+                    }
+                }
+            })
+            .unwrap();
+
+        let workspace_root: &Path = self
+            .build_cmd
+            .manifest_path
+            .parent()
+            .unwrap_or_else(|| Path::new("."));
+        let env_toml = workspace_root.join("environments.toml");
+        if env_toml.exists() {
```

Comment:
In addition to noticing if you _modify_ `environments.toml`, can we also check if you _add_ or _delete_ it?

Reply:
I modified the code to do this. It was a pretty substantial change because often `environments.toml` is in a  directory not otherwise watched. To do this I had to watch its parent directory but then make sure only changes to that file were made (assuming its not in a directory already being watched for changes). I used Arc<PathBuf> and Arc::clone to do this because the lifetimes of the paths being passed to the watcher function were causing me issues.